mist-common/src/main/java/edu/snu/mist/common/operators/ApplyStatefulOperator.java

```diff
@@ -49,6 +51,12 @@
    */
   private long latestCheckpointTimestamp;
 
+  /**
+   * The map of states for checkpointing.
+   * The key is the time of the checkpoint event, and the value is the state of this operator at that timestamp.
+   */
+  protected Map<Long, Map<String, Object>> checkpointMap;
```

Comment:
Why do you put Map<String, Object>, instead of `Object`?

Reply:
Each state of an operator is consisted of multiple objects.
One state is checkpointed as a Map<String, Object>.
I don't think it should necessarily be an Object.

Reply:
In line 98, you create `stateMap`. I'm not sure creating the map is necessary. Could you please explain why we need the map?

Reply:
The term `stateMap` may be confusing. A `stateMap` is actually the operator's current state.
For stateful operators that have only one object in the `stateMap`, there may be no need a map.
But some stateful operators have multiple objects for the `stateMap`, such as `SessionWindowOperator`.
For the sake of conformity, I think usage of the map can be justified.